File: app/src/main/java/com/artmurka/artmurkaapp/Model/Modules/WishListRequest.java
package com.artmurka.artmurkaapp.Model.Modules;

import com.artmurka.artmurkaapp.Model.InterfacesModel.IWishList;
import com.artmurka.artmurkaapp.Model.Pojo.ItemList.WishList.WishList;

import java.util.HashMap;

import retrofit2.Call;


public class WishListRequest implements IWishList {

    @Override
    public Call<WishList> toWishList(String goods_id) {
        UcozApiModule ucoz = new UcozApiModule();

        HashMap<String, String> mapForUcozModule = new HashMap<String, String>();
        mapForUcozModule.put("goods_id", goods_id);
        HashMap<String, String> confForRequest = ucoz.get("POST","uapi/shop/wishlisth",mapForUcozModule);

        return ApiModule.getClient().addToWishList(confForRequest);
//                .subscribeOn(Schedulers.newThread())
//                .observeOn(AndroidSchedulers.mainThread());
    }

    @Override
    public Call<WishList> getWishList() {

        UcozApiModule ucoz = new UcozApiModule();
        HashMap<String, String> mapForUcozModule = new HashMap<String, String>();
        mapForUcozModule.put("page", "wishlist");
        HashMap<String, String> confForRequest = ucoz.get("GET","uapi/shop/request",mapForUcozModule);
        return ApiModule.getClient().getWishList(confForRequest);
    }
}

File: app/src/main/java/com/artmurka/artmurkaapp/data/model/modules/WishListRequest.kt
package com.artmurka.artmurkaapp.data.model.modules

import com.artmurka.artmurkaapp.data.model.interfacesmodel.IWishList
import com.artmurka.artmurkaapp.data.model.pojo.itemlist.wishList.WishList
import com.artmurka.artmurkaapp.data.model.retrofit.ApiRetrofit
import io.reactivex.Observable
import java.util.HashMap
import javax.inject.Inject


class WishListRequest  @Inject constructor(val apiModule: ApiRetrofit, val ucoz : UcozApiModule ): IWishList {

    override val wishList: Observable<WishList>
        get() {
            val mapForUcozModule = HashMap<String, String>()
            mapForUcozModule["page"] = "wishlist"
            val confForRequest = ucoz["GET", "uapi/shop/request", mapForUcozModule]
            return apiModule.getWishList(confForRequest)
        }

    override fun toWishList(goods_id: String): Observable<WishList> {
        val mapForUcozModule = HashMap<String, String>()
        mapForUcozModule["goods_id"] = goods_id
        val confForRequest = ucoz["POST", "uapi/shop/wishlisth", mapForUcozModule]

        return apiModule.addToWishList(confForRequest)
        //                .subscribeOn(Schedulers.newThread())
        //                .observeOn(AndroidSchedulers.mainThread());
    }
}
